src/App.js

```diff
@@ -1,9 +1,40 @@
 import "./App.css";
-import logo from "./logo.png";
+import 'react-toastify/dist/ReactToastify.css';
+import { ToastContainer } from 'react-toastify';
+import { Navbar, Container } from 'component';
+import { Login, Signup, Home } from 'features';
+import { useSelector } from 'react-redux';
+import { ProtectedRoute } from "routes/ProtectedRoute";
+import { Route, Routes } from 'react-router-dom';
 
 function App() {
+  const token = useSelector(state => state.auth.token);
```

Comment:
Suggestion - you might have to use the token, as well as other auth states in other places in your app. So instead of writing the callback everytime, you could export it from your slice, import where you want to access auth states and pass it to useSelector.

Reply:
Make sense I'll export it from the slice itself